### PR TITLE
More tidy ups

### PR DESCRIPTION
**Pull request by huoyaoyuan**

### CODE REVIEW

File: osu.Game/Online/Chat/Message.cs [hunk 1]
```diff
@@ -42,17 +42,9 @@ public Message(long id)
...
-
-            return Id == objMessage?.Id;
-        }
+        public int CompareTo(Message other) => Id.CompareTo(other.Id);
 
-        public override int GetHashCode()
-        {
-            return Id.GetHashCode();
-        }
+        public bool Equals(Message other) => Id == other?.Id;
```
Comment (peppy) [inline osu.Game/Online/Chat/Message.cs]: have you tested this still works with chat de-duping?

**Review by smoogipoo — DISMISSED** [review]
Tested, works fine with chat deduping. 👍

**Comment by huoyaoyuan** [comment]
There are so many things can be tidy
*See the power of Linq*

File: osu.Game/Online/Chat/Message.cs (hunk 1 above)
Comment (huoyaoyuan) [inline osu.Game/Online/Chat/Message.cs]: it seems not working well...looking about reasons

Comment (huoyaoyuan) [inline osu.Game/Online/Chat/Message.cs]: the generic `Equals` hasn't got called at all...weird

Comment (huoyaoyuan) [inline osu.Game/Online/Chat/Message.cs]: OK, after a reading of referenced source, I found that `GetHashCode` is necessary for linq `Except`, as it uses an hash set. Two element with `Equals == true` won't be treated equal if they are in different buckets.
Now it should work.

**Review by peppy — APPROVED** [review]

**Pull request merged**